src/Generating/FinishGen.cpp

```diff
@@ -1840,7 +1840,7 @@ cFinishGenOres::OreInfos cFinishGenOres::OreInfosFromString(const AString & a_Or
 			LOGWARNING("Cannot parse ore information from string, invalid number in OreInfo \"%s\".", ore.c_str());
 			continue;
 		}
-		res.emplace_back(oreType, oreMeta, maxHeight, numNests, nestSize);
+		res.emplace_back(static_cast<BLOCKTYPE>(oreType), oreMeta, maxHeight, numNests, nestSize);
```

Comment:
Probably better to change BlockStringToType here so that it returns BlockType, rather than casting the result...

Reply:
Actually, The function may return -1 if the block is unknown and the data type blew everything up. I already tried this

Reply:
`BLOCKTYPE` is `unsigned char` so no way we get a -1 in there

Reply:
Ugh, C-style brittleness... If only strong type systems had caught on sooner. Nevermind, let's just mark it down as another place to use optional types once we get around to that conversion effort.